integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/ChangeDataTypeTestCases.scala

```diff
@@ -164,6 +164,21 @@ class ChangeDataTypeTestCases extends Spark2QueryTest with BeforeAndAfterAll {
     sql("drop table if exists PreAggMain_preagg1")
   }
 
+  test("test data type change for dictionary exclude INT type column") {
```

Comment:
PLease add tests for short to int, short to long and float to double also

Reply:
Alter table datatype change does not support short to int, short to long and float to double. It only supports Int to BigInt and decimal from lower precision to higher precision. If any other data type change request comes validation is done and exception is thrown